thor/tests/e2e.rs

```diff
@@ -5,397 +5,334 @@ mod harness;
 use anyhow::Result;
 use bitcoin::Amount;
 use bitcoin_harness::{self, Bitcoind};
+use futures::future;
 use genawaiter::GeneratorState;
-use harness::{build_runtime, generate_balances, make_transports, make_wallets, Transport};
+use harness::{generate_balances, make_transports, make_wallets, Transport, Wallet};
+use spectral::prelude::*;
+use testcontainers::clients::Cli;
 use thor::{Balance, Channel, PtlcPoint, PtlcSecret};
 
-#[test]
-fn e2e_channel_creation() {
-    let mut runtime = build_runtime();
-
-    let tc_client = testcontainers::clients::Cli::default();
-    let bitcoind = Bitcoind::new(&tc_client, "0.19.1").unwrap();
-    runtime.block_on(bitcoind.init(5)).unwrap();
-
-    let fund_amount_alice = Amount::ONE_BTC;
-    let fund_amount_bob = Amount::ONE_BTC;
-    let (balance_alice, balance_bob) = generate_balances(fund_amount_alice, fund_amount_bob);
-
+// NOTE: For some reason running these tests overflows the stack. In order to
+// mitigate this run them with:
+//
+//     RUST_MIN_STACK=10000000 cargo test
+
+async fn create_channels(
+    bitcoind: &Bitcoind<'_>,
+) -> (
+    Amount,
```

Comment:
I'd expect the fund amount to be passed (even if it's always same value) instead of being returned.
Or not even return it if it's always the same.

Reply:
I added a `const`, makes it cleaner. Thanks